cmd/lncli/invoicesrpc_active.go

```diff
@@ -38,8 +38,11 @@ var settleInvoiceCommand = cli.Command{
 	Usage:    "Reveal a preimage and use it to settle the corresponding invoice.",
 	Description: `
 	Todo.`,
-	ArgsUsage: "preimage",
 	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "hash",
```

Comment:
imo this should be settled by payment address, as that will be able to handle AMP or the existing keysend variant. settling by payment hash is only workable for single-shard keysends.

in moving to AMP we will being moving to payment address as the primary index for settling invoices anyway.

Reply:
There are always still legacy (hodl) invoices with payment address. I am not too convinced that that dependency really exists.

Reply:
> There are always still legacy (hodl) invoices with payment address.

I'm not sure I follow, those would be settled via a preimage.

Reply:
This PR also allows regular invoices to be handled as if they were hodl invoices. For example for a webshop inventory check. In that case, the caller doesn't need to have the preimage to settle, because it is already in `lnd`'s database.